src/mame/midway/midyunit_m.cpp

```diff
@@ -583,32 +562,54 @@ void midyunit_state::init_totcarn()
  *
  *************************************/
 
-MACHINE_RESET_MEMBER(midyunit_state,midyunit)
+void midyunit_base_state::machine_start()
 {
-	// reset sound
-	switch (m_chip_type)
-	{
-		case SOUND_NARC:
-			m_narc_sound->reset_write(1);
-			m_narc_sound->reset_write(0);
-			break;
+	// allocate memory
+	m_cmos_ram = std::make_unique<uint16_t[]>((0x2000 * 4)/2);
+	m_nvram->set_base(m_cmos_ram.get(), 0x2000 * 4);
 
-		case SOUND_CVSD:
-		case SOUND_CVSD_SMALL:
-			m_cvsd_sound->reset_write(1);
-			m_cvsd_sound->reset_write(0);
-			break;
+	// reset all the globals
+	m_cmos_page = 0;
 
-		case SOUND_ADPCM:
-			m_adpcm_sound->reset_write(1);
-			m_adpcm_sound->reset_write(0);
-			break;
+	save_item(NAME(m_cmos_page));
+	save_item(NAME(m_prot_result));
+	save_item(NAME(m_prot_sequence));
+	save_item(NAME(m_prot_index));
+	save_item(NAME(m_cmos_w_enable));
+	save_pointer(NAME(m_cmos_ram), (0x2000 * 4)/2);
+}
 
-		case SOUND_YAWDIM:
-			break;
-	}
+void term2_state::machine_start()
+{
+	midyunit_base_state::machine_start();
+
+	m_left_flash.resolve();
+	m_right_flash.resolve();
+	m_left_gun_recoil.resolve();
+	m_right_gun_recoil.resolve();
+	m_left_gun_green_led.resolve();
+	m_left_gun_red_led.resolve();
+	m_right_gun_green_led.resolve();
+	m_right_gun_red_led.resolve();
 }
 
+void midzunit_state::machine_reset()
+{
+	m_narc_sound->reset_write(1);
+	m_narc_sound->reset_write(0);
+}
+
+void midyunit_cvsd_state::machine_reset()
+{
+	m_cvsd_sound->reset_write(1);
+	m_cvsd_sound->reset_write(0);
+}
+
+void midyunit_adpcm_state::machine_reset()
+{
+	m_adpcm_sound->reset_write(1);
+	m_adpcm_sound->reset_write(0);
+}
```

Comment:
Can you have these call `midyunit_base_state::machine_reset()` in case it has something important added?